### DISCUSSION TITLE
New versioning schema?

### DISCUSSION THREAD

**Discussion by ezloj** [renovatebot/renovate#14868] [Suggest an Idea]
https://github.com/renovatebot/renovate/tree/main/lib/modules/versioning includes a number of popular and widely used versioning schemas. Would it be acceptable to add a new one there?

Specifically, my organisation uses a versioning pattern for docker images that consists of several semantic versions. So a full image version could look like this:
`1.0.0-2.0.0-3.0.0` where the first part, the 1.0.0 is the version of a main image component and other parts describe the versions for other major components in an image. To make it work in renovate, I'd need to add logic to the comparison functions that would recursively read the sub-semantic versions and compare them from left to right with the left-most being the most important and right-most being the least. E.g.:
1.0.0-2.0.0 < 1.0.0-2.0.1
1.0.0-2.0.0-3.1.0 < 1.0.0-2.0.1-4.0.0

Would me raising a pull request that adds such versioning system to the list of the ones already described by renovate be accepted if working properly and not affecting anything else in renovate?

**Comment by rarkins** [comment]
As a general rule we don't like to support proprietary modules which are useful to only one or a very small subset of end users. Does `loose` versioning already work?

One challenge though is how to classify updates as major/minor/patch

**Reply by ezloj** [reply]
I will test the `loose` versioning. I didn't notice it when reading the documentation. Will need some testing though. Thank you.

**Comment by ezloj** [comment]
`loose` versioning partially solves the issue. Enough to stop considering a new versioning schema. Thank you @rarkins
